From: =?utf-8?b?TmVsc29uIEJlbsOtdGV6IExlw7Nu?= <nbenitezl@gmail.com>
Date: Sun, 19 Apr 2020 14:16:02 -0400
Subject: FileChooserButton: query 'display name' also for unbookmarked files

Do also the async file info query for remote files when they
are not bookmarked, because otherwise "None" will be shown as
file name (and fallback text generic icon will be used).

The remote file was already browsed by the file chooser
instance when selecting it, so querying the display name
again should be using gio cache and not be slow.

Even if it's were slow it's better than showing 'None'
which makes it seem as if nothing was selected.

Bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/1966
Origin: upstream, 3.24.31, commit:b01f371ce61f8077bb70a2b6024f4163fafffc5c
---
 gtk/gtkfilechooserbutton.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkfilechooserbutton.c b/gtk/gtkfilechooserbutton.c
index fa9b945..8711385 100644
--- a/gtk/gtkfilechooserbutton.c
+++ b/gtk/gtkfilechooserbutton.c
@@ -2550,7 +2550,8 @@ update_label_and_image (GtkFileChooserButton *button)
 	    }
         }
 
-      if (g_file_is_native (file))
+      if (g_file_is_native (file) ||
+          !_gtk_bookmarks_manager_has_bookmark (button->priv->bookmarks_manager, file))
         {
           priv->update_button_cancellable =
             _gtk_file_system_get_info (priv->fs, file,
